controller/token_blackbox_test.go

```diff
@@ -98,6 +99,35 @@ func (rest *TestTokenREST) TestRefreshTokenUsingNilTokenFails() {
 	assert.NotNil(t, err)
 }
 
+func (rest *TestTokenREST) TestRefreshTokenUsingWrongRefreshTokenFails() {
+	t := rest.T()
+	rest.exchangeStrategy = "401"
+	service, controller := rest.SecuredController()
+
+	refreshToken := "WRONG_REFRESH_TOKEN"
+	payload := &app.RefreshToken{
+		RefreshToken: &refreshToken,
+	}
+	test.RefreshTokenUnauthorized(t, service.Context, service, controller, payload)
+}
+
+func (rest *TestTokenREST) TestRefreshTokenUsingCorrectRefreshTokenOK() {
+	t := rest.T()
+	service, controller := rest.SecuredController()
+
+	refreshToken := "SOME_REFRESH_TOKEN"
+	payload := &app.RefreshToken{
+		RefreshToken: &refreshToken,
+	}
+	_, authToken := test.RefreshTokenOK(t, service.Context, service, controller, payload)
+	token := authToken.Token
+	require.NotNil(rest.T(), token.TokenType)
+	require.Equal(rest.T(), "bearer", *token.TokenType)
+	require.NotNil(rest.T(), token.AccessToken)
+	require.Equal(rest.T(), rest.sampleAccessToken, *token.AccessToken)
+	require.NotNil(rest.T(), token.RefreshToken)
+	require.Equal(rest.T(), rest.sampleRefreshToken, *token.RefreshToken)
```

Comment:
Check the expire_in too.
And move all these checks to a new function and reuse it in both TestRefreshTokenUsingCorrectRefreshTokenOK() and checkExchangeWithRefreshToken()
```
	require.NotNil(rest.T(), token.TokenType)
 	require.Equal(rest.T(), "bearer", *token.TokenType)
 	require.NotNil(rest.T(), token.AccessToken)
 	require.Equal(rest.T(), rest.sampleAccessToken, *token.AccessToken)
 	require.NotNil(rest.T(), token.RefreshToken)
 	require.Equal(rest.T(), rest.sampleRefreshToken, *token.RefreshToken)
 	expiresIn, err := strconv.Atoi(*token.ExpiresIn)
 	require.Nil(rest.T(), err)
 	require.True(rest.T(), expiresIn > 60*59*24*30 && expiresIn < 60*61*24*30) // The expires_in should be withing a minute range of 30 days.
```

Reply:
OK, I see that these are two different types, so, it's tricky to re-use the same function. And I've added the check for expireIn.